### DISCUSSION TITLE
when dismissing using esc key, should the focus be visible?

### DISCUSSION THREAD

**Discussion by herzaso** [floating-ui/floating-ui#2806] [React]
I have a menu (floating-ui) that is opened with a click of a button.
When I click on any option, the menu is closed and the focus is not visible.
However, when I click on ESC key, the menu is dismissed and the focus is visible on the button trigger.

Is that intentional? Why is that?

**Comment by atomiks** [comment]
Yes, it's how the browser determines the `:focus-visible` state. The focus is visible only while using the keyboard.

**Reply by herzaso** [reply]
Something still looks funny.
I have a modal dialog with these behaviors:

If the focus is not visible and I click on the trigger button, the modal opens and the modal's close button has a visible focus (no keyboard key used). When I click on the close button, the trigger has no visible focus.

In another scenario, I use TAB to show the focus on the trigger. I then click on the trigger, the modal's close button has a visible focus, I click on the close button, the focus on the trigger is visible again.
In this situation, I can't hide the visible focus (even if I move the focus from the trigger, then click the trigger and close the modal, the focus is again visible)

Another thing - these scenarios happen when I click on the modal's close button. If I click outside of the modal, the modal closes and I don't see the focus.

I still don't understand if I'm doing something wrong 🤔

**Reply by herzaso** [reply]
@atomiks, did you see my update on this issue? Is it still valid?
